src/ceph-daemon/ceph-daemon

```diff
@@ -296,30 +296,34 @@ def check_unit(unit_name):
         state = 'unknown'
     return (enabled, state)
 
-def get_legacy_config_fsid(cluster):
-    try:
-        config = ConfigParser()
-        config.read('/etc/ceph/%s.conf' % cluster)
-        if config.has_section('global') and config.has_option('global', 'fsid'):
-            return config.get('global', 'fsid')
-    except Exception as e:
-        logger.warning('unable to parse \'fsid\' from \'[global]\' section of /etc/ceph/ceph.conf: %s' % e)
-        return None
+def get_legacy_config_fsid(cluster, base_dir=None):
+    config_file = '/etc/ceph/%s.conf' % cluster
+    if base_dir is not None:
+        config_file = os.path.abspath(base_dir + config_file)
```

Comment:
i think all these abspath calls should be os.path.join(base_dir, ...)

Reply:
I thought that too, but found `os.path.join` will discard all prior components if any subsequent component is an absolute path.

https://docs.python.org/2/library/os.path.html#os.path.join

```
# python3
Python 3.6.9 (default, Sep 09 2019, 18:26:39) [GCC] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import os
>>> base_dir = '/foo'
>>> config_file = '/etc/ceph/ceph.conf'
>>> os.path.join(base_dir, config_file)
'/etc/ceph/ceph.conf'
>>> os.path.abspath(base_dir + config_file)
'/foo/etc/ceph/ceph.conf'
```